### ISSUE TITLE
Classes missing during test

### ISSUE THREAD

**Issue by raboof** [lightbend/genjavadoc#209]
Since #208, when running the tests, there are sometimes classes that are missing at run or compile time.

It looks like this is due to our overridden `test` task:

```
    test in Test := {
      // since we are building for different Scala patch versions, a clean
      // is required to avoid conflicts in class files
      clean.value
      saveTestClasspath.value
      (test in Test).value
    },
```

It looks like sometimes the `clean` removes class files (as intended), but `test in Test` doesn't correctly re-introduce it by compiling it again.

However, when commenting out that line, it looks like sbt already recompiles the classes when switching to a different Scala patchversion. The `clean.value` is perhaps no longer needed?

I think the `clean.value` was introduced in https://github.com/lightbend/genjavadoc/commit/cddacd1a7427cc0c5a7577e3d39c0c8609d68bb2 , @jodersky do you know how we could verify whether this is still needed?

**Comment by SethTisue** [comment]
fixed by #210. if there's trouble as a result, let's cross that bridge when we come to it. perhaps this was only necessary in some old version of sbt?

regardless, curious if @jodersky remembers anything about this

**Comment by jodersky** [comment]
Hey, sorry I'm not up-to-date with this anymore
